src/components/usercomponents/NotificationPanel.tsx

```diff
@@ -0,0 +1,188 @@
+    };
+  }, []);
+
+  useEffect(() => {
+    socket.emit('fetchNotifications', userId);
+
+    socket.on('notifications', (data) => {
+      dispatch(setNotifications(data));
+      setIsLoading(false);
+    });
+
+    socket.on('error', (error) => {
+      setError(error);
+      setIsLoading(false);
+    });
+
+    return () => {
+      socket.off('notifications');
+      socket.off('error');
+    };
+  }, [dispatch]);
+
+  const togglePanel = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const toggleExpand = async (id: string) => {
+    const notification = notifications.find((n) => n.id === id);
+
+    if (!notification || notification.isRead) {
+      setExpandedNotificationId(expandedNotificationId === id ? null : id);
+      return;
+    }
+
+    setLoadingNotifications((prev) => [...prev, id]);
+    try {
+      await markAsRead(id).unwrap();
+      dispatch(markAsReadInStore(id));
+    } catch (error) {
+      toast.error('Error while marking notification as read');
+    }
+
+    setLoadingNotifications((prev) =>
+      prev.filter((notificationId) => notificationId !== id),
+    );
+    setExpandedNotificationId(expandedNotificationId === id ? null : id);
+  };
+
+  const sortedNotifications = [...notifications].sort(
+    (a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime(),
+  );
+
+  const filteredNotifications = sortedNotifications.filter((notification) => {
+    if (filter === 'unread') {
+      return !notification.isRead;
+    } else if (filter === 'today') {
+      const today = new Date().toLocaleDateString();
+      const notificationDate = new Date(
+        notification.createdAt,
+      ).toLocaleDateString();
+      return notificationDate === today;
+    }
+    return true;
+  });
+
+  return (
+    <div className="relative">
+      <ToastContainer />
+      <Button onClick={togglePanel}>
+        <FaBell className="h-6 w-6" />
+        {notifications.some((notification) => !notification.isRead) && (
+          <NotificationBadge
+            count={
+              notifications.filter((notification) => !notification.isRead)
+                .length
+            }
+          />
+        )}
+      </Button>
+      {isOpen && (
+        <>
+          <div
+            ref={notificationPanelRef}
+            className="absolute right-0 mt-2 w-80 bg-white border border-gray-200 shadow-lg rounded-lg z-20 overflow-hidden"
+          >
+            <div className="p-4">
+              <p className="text-gray-700 font-semibold mb-3">Notifications</p>
+              <NotificationFilters filter={filter} setFilter={setFilter} />
+              {isLoading ? (
+                <div className="space-y-4">
+                  {[...Array(5)].map((_, index) => (
+                    <div
+                      key={index}
+                      className="animate-pulse flex space-x-4 p-4"
+                    >
+                      <div className="rounded-full bg-gray-300 h-12 w-12"></div>
```

Comment:
```suggestion
                      <div className="rounded-full bg-gray-300 h-12 w-12" />
```

Reply:
Can you work on this instead of resolving it before you do it? What's so difficult her?